artifactory.py

```diff
@@ -1897,6 +1903,16 @@ def lchmod(self, mode):
         """
         raise NotImplementedError()
 
+    def unlink(self, missing_ok=False):
+        """
+        Removes a file or folder
+        """
+        try:
+            self._accessor.unlink(self)
+        except ArtifactoryException:
+            if not missing_ok:
```

Comment:
In `missing` case accessor gives OSError, not ArtifactoryException 
https://github.com/devopshq/artifactory/blob/master/artifactory.py#L1000-L1001

ArtifactoryException shows some permissions issues and should be raised

Reply:
But `OSError` should be `FileNotFoundError`, as an original Path raises btw

Reply:
Thanks for your advice, I have changed the exception to FileNotFoundError for unlink.